package/MDAnalysis/core/topologyobjects.py

```diff
@@ -764,14 +764,50 @@ def __repr__(self):
 
     def __eq__(self, other):
         """Test if contents of TopologyGroups are equal"""
-        return (self.indices == other.indices).all()
+        return np.array_equal(self.indices, other.indices)
 
     def __ne__(self, other):
         return not self == other
 
     def __nonzero__(self):
         return not len(self) == 0
 
+    @property
+    def atom1(self):
+        """The first atom in each TopologyObject in this Group"""
+        return self._ags[0]
+
+    @property
+    def atom2(self):
+        """The second atom in each TopologyObject in this Group"""
+        return self._ags[1]
+
+    @property
+    def atom3(self):
+        """The third atom in each TopologyObject in this Group"""
+        try:
+            return self._ags[2]
+        except IndexError:
+            nvert = {'bond': 2,
+                     'angle': 3,
+                     'dihedral': 4,
+                     'improper': 4}[self.btype]
+            raise IndexError("TopologyGroup of {}s only has {} vertical AtomGroups"
+                             "".format(self.btype, nvert))
+
+    @property
+    def atom4(self):
+        """The fourth atom in each TopologyObject in this Group"""
+        try:
+            return self._ags[3]
+        except IndexError:
+            nvert = {'bond': 2,
+                     'angle': 3,
+                     'dihedral': 4,
+                     'improper': 4}[self.btype]
+            raise IndexError("TopologyGroup of {}s only has {} vertical AtomGroups"
+                             "".format(self.btype, nvert))
+
```

Comment:
It's mostly for the distance calculations.  It just transposes the bond records into vertical slices which is what you want 

```
[(a1, a2, a3), (b1, b2, b3), ...] -> [(a1, b1, ...), (a2, b2, ...), (a3, b3, ...)]
```

So bond length calculation is something like `norm(atom1.positions - atom2.positions)`